cpu/atmega2560/reboot_arch.c

```diff
@@ -14,27 +15,21 @@
  * @brief       Implementation of the kernels reboot interface
  *
  * @author      Hinnerk van Bruinehsen <h.v.bruinehsen@fu-berlin.de>
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
  *
  * @}
  */
 
-#include <stdio.h>
 #include <avr/wdt.h>
 
-#include "arch/reboot_arch.h"
+#include "kernel.h"
 #include "cpu.h"
 
-
-int reboot_arch(int mode)
+void reboot(void)
 {
-    printf("Going into reboot, mode %i\n", mode);
-
     /*
      * Since the AVR doesn't support a real software reset, we set the Watchdog
      * Timer on a 250ms timeout. Consider this a kludge.
      */
     wdt_enable(WDTO_250MS);
-
```

Comment:
shouldn't there be an `while(1);` loop after this, so the execution stays here until the watchdog times out?